### DISCUSSION TITLE
Extract changelogs without repo access

### DISCUSSION THREAD

**Discussion by HenrikPoulsen** [renovatebot/renovate#10900] [Request Help]
Hello,
So on our private github enterprise instance I have setup renovate on a handful repos so far, to much rejoicing.
This renovate is extended with datasources and managers for our internal dependency systems.

Renovate is setup to run in each repo, so people get maximum control over run frequency and they can decide when to merge updates to our docker image that we ship our extended renovate with (renovate renovating renovate is so nice btw :) )

We are big on the principle of least priviledge here so when renovate runs in most cases it only has access to the repo it is running inside as well as our `renovate-config` repo in the org.
This will obviously create problems for changelog extraction since it seems to be gitlab/github based, and renovate will not have arbitrary read access in the org (don't want to risk leaking tokens with that much power so they are highly scoped).

I was wondering if there is perhaps some easy way that we could add a mechanism to query the changelog from an internal website instead of using github credentials to get them (just for our own managers/datasources).

We have so far survived by just pulling renovate in as a npm dependency so we have not had to fork the repo, just to keep conflics low and allows us to have high automation here.
It does look to me like the code would clobber release notes that a manager might already have put in there using other means.
Is there some means to be able to allowing a custom manager or similar with custom changelog fetching logic and have it not go down and not be overwritten by the gitlab/github logic?

**Comment by rarkins** [comment]
Our changelog logic is generalized - not per manager - and in big need of a rewrite. When that happens, one of the things we'll want to add is the ability to pull and parse arbitrary changelog URLs. If that was implemented, you could perhaps have a dedicated repo whose purpose is to run a GitHub Action regularly, pull release notes from all your repos and then centralize them into markdown files in the one repo.

If you didn't need *embedded* release notes and instead would be satisfied with a link to somewhere, then that's much easier to implement.

**Reply by HenrikPoulsen** [reply]
Thanks for the response. And sorry about the slow reply. Was on vacation for a bit :)
Adding links would be be a reasonable workaround.
Do you have any pointers what the "correct" way of doing that would be?

**Reply by rarkins** [reply]
You could do it by adding a new table column and contents for certain repos using `prBodyColumns` or having it added below the table using `prBodyNotes`

**Reply by HenrikPoulsen** [reply]
Thanks. I'll do that
